types/react-transition-group/index.d.ts

```diff
@@ -6,7 +6,34 @@
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 // TypeScript Version: 2.8
 
-export { default as Transition } from './Transition';
-export { default as CSSTransition } from './CSSTransition';
-export { default as TransitionGroup } from './TransitionGroup';
-export { default as SwitchTransition } from './SwitchTransition';
+export {
+    default as Transition,
+    EndHandler,
+    EnterHandler,
+    ExitHandler,
+    TransitionActions,
+    TransitionChildren,
+    TransitionProps,
+    TransitionStatus,
+    UNMOUNTED,
+    EXITED,
+    ENTERING,
+    ENTERED,
+    EXITING,
+} from './Transition';
+export {
+    default as CSSTransition,
+    CSSTransitionClassNames,
+    CSSTransitionProps,
+} from './CSSTransition';
+export {
+    default as TransitionGroup,
+    IntrinsicTransitionGroupProps,
+    ComponentTransitionGroupProps,
+    TransitionGroupProps,
+} from './TransitionGroup';
+export {
+    default as SwitchTransition,
+    modes,
```

Comment:
Does `export modes` work properly? Because `export modes` is not included in [the original `index.js`](https://github.com/reactjs/react-transition-group/blob/b9b43f9e37615f7d4aa05f0770a89c13b57bae4f/src/index.js#L1).

```js
import { modes } from 'react-transition-group/SwitchTransition';
//=> OK

import { modes } from 'react-transition-group';
=> ???
```

Please see also [the original `SwitchTransition.js`](https://github.com/reactjs/react-transition-group/blob/b9b43f9e37615f7d4aa05f0770a89c13b57bae4f/src/SwitchTransition.js#L23
).